DynamicTablesPkg/Library/Common/TableHelperLib/ConfigurationManagerObjectParser.c

```diff
@@ -893,6 +894,7 @@ STATIC CONST CM_OBJ_PARSER_ARRAY  ArchCommonNamespaceObjectParser[] = {
   CM_PARSER_ADD_OBJECT (EArchCommonObjFixedFeatureFlags,            CmArchCommonFixedFeatureFlagsParser),
   CM_PARSER_ADD_OBJECT (EArchCommonObjCmRef,                        CmArchCommonObjRefParser),
   CM_PARSER_ADD_OBJECT (EArchCommonObjPciConfigSpaceInfo,           CmArchCommonPciConfigSpaceInfoParser),
+  CM_PARSER_ADD_OBJECT (EArchCommonObjMcfgPciConfigSpaceInfo,       CmArchCommonPciConfigSpaceInfoParser),

```

Comment:
The structure is an array, so the location is important. I think the new entry should be added at the end.
A parser for the `EArchCommonObjTpm2DeviceInfo` object is missing, but it can be added afterwards I guess.